### PR TITLE
Add table gcp_bigquery_table closes #191

### LINKED ISSUE #191: Add table gcp_bigquery_table
**References**
https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#Table

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/gcp_bigquery_table []

PRETEST: tests/gcp_bigquery_table

TEST: tests/gcp_bigquery_table
Running terraform
data.google_client_config.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
google_bigquery_dataset.named_test_resource: Creating...
google_service_account.named_test_resource: Creating...
google_bigquery_dataset.named_test_resource: Creation complete after 4s [id=projects/pikachu-aaa/datasets/turbottest53721]
google_bigquery_table.named_test_resource: Creating...
google_service_account.named_test_resource: Creation complete after 5s [id=projects/pikachu-aaa/serviceAccounts/turbottest53721@pikachu-aaa.iam.gserviceaccount.com]
google_bigquery_table.named_test_resource: Creation complete after 4s [id=projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

etag = dOVVbzh7xzv5w6CQOevhLw==
project_id = pikachu-aaa
region_id = us-east1
resource_aka = gcp://bigquery.googleapis.com/projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721
resource_id = projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721
resource_name = turbottest53721
self_link = https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721
service_account_email = turbottest53721@pikachu-aaa.iam.gserviceaccount.com

Running SQL query: test-get-query.sql
[
  {
    "dataset_id": "turbottest53721",
    "id": "pikachu-aaa:turbottest53721.turbottest53721",
    "kind": "bigquery#table",
    "labels": {
      "name": "turbottest53721"
    },
    "location": "us-east1",
    "project": "pikachu-aaa",
    "range_partitioning": null,
    "self_link": "https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721",
    "table_id": "turbottest53721",
    "time_partitioning": {
      "type": "DAY"
    },
    "title": "turbottest53721",
    "type": "TABLE"
  }
]
✔ PASSED

Running SQL query: test-invalid-name-query.sql
null
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "dataset_id": "turbottest53721",
    "table_id": "turbottest53721"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://bigquery.googleapis.com/projects/pikachu-aaa/datasets/turbottest53721/tables/turbottest53721"
    ],
    "tags": {
      "name": "turbottest53721"
    },
    "title": "turbottest53721"
  }
]
✔ PASSED

POSTTEST: tests/gcp_bigquery_table

TEARDOWN: tests/gcp_bigquery_table

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic info

```sql
select
  table_id,
  dataset_id,
  self_link,
  creation_time,
  location
from
  gcp_bigquery_table;
```
```
+--------------+----------------+--------------------------------------------------------------------------------------------------------------+---------------------+----------+
| table_id     | dataset_id     | self_link                                                                                                    | creation_time       | location |
+--------------+----------------+--------------------------------------------------------------------------------------------------------------+---------------------+----------+
| test_table_1 | test_dataset_1 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/datasets/test_dataset_1/tables/test_table_1 | 2021-05-13 05:08:41 | US       |
+--------------+----------------+--------------------------------------------------------------------------------------------------------------+---------------------+----------+
```

### List tables which are not encrypted using CMK

```sql
select
  table_id,
  dataset_id,
  location,
  kms_key_name
from
  gcp_bigquery_table
where
  kms_key_name is null;
```
```
+--------------+----------------+----------+--------------+
| table_id     | dataset_id     | location | kms_key_name |
+--------------+----------------+----------+--------------+
| test_table_1 | test_dataset_1 | US       | <null>       |
+--------------+----------------+----------+--------------+
```

### List tables which do not have owner tag key

```sql
select
  dataset_id,
  location
from
  gcp_bigquery_table
where
  tags -> 'owner' is null;
```
```
+----------------+----------+
| dataset_id     | location |
+----------------+----------+
| test_dataset_1 | US       |
+----------------+----------+
```

</details>

### CODE REVIEW

File: gcp/plugin.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 		TableMap: map[string]*plugin.Table{
 			"gcp_audit_policy":                    tableGcpAuditPolicy(ctx),
 			"gcp_bigquery_dataset":                tableGcpBigQueryDataset(ctx),
+			"gcp_bigquery_table": 				   tableGcpBigqueryTable(ctx),
```
Comment (Subhajit97) [inline gcp/plugin.go:33]: Fix indentation

File: gcp/table_gcp_bigquery_table.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+			KeyColumns: plugin.AllColumns([]string{"dataset_id", "id"}),
+			Hydrate:    getBigqueryTable,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listBigQueryDatasets,
+			Hydrate:       listBigqueryTables,
+		},
+		Columns: []*plugin.Column{
+			// commonly used columns
+			{
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:27]: Can we add the important columns at the top, i.e. `table_id`, `dataset_id`, `self_link`, `id`?

File: gcp/table_gcp_bigquery_table.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+				Name:        "expiration_time",
+				Description: "The time when this table expires, in milliseconds since the epoch. If not present, the table will persist indefinitely. Expired tables will be deleted and their storage reclaimed. The defaultTableExpirationMs property of the encapsulating dataset can be used to set a default expirationTime on newly created tables.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("ExpirationTime").Transform(transform.UnixMsToTimestamp),
+			},
+			{
+				Name:        "friendly_name",
+				Description: "A descriptive name for this table.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("FriendlyName"),
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:49]:
```suggestion
```

File: gcp/table_gcp_bigquery_table.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getBigqueryTable,
+			},
+			{
+				Name:        "view",
+				Description: "dditional details for a view.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("View"),
+			},
+			// standard steampipe columns
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:105]:
```suggestion
			// Steampipe standard columns
```

File: gcp/table_gcp_bigquery_table.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("TableReference.ProjectId"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:142]:
```suggestion
//// LIST FUNCTION

```

File: gcp/table_gcp_bigquery_table.go (hunk 2 above)
Comment (ParthaI) [reply gcp/table_gcp_bigquery_table.go:27]: We are sorting the columns in alphabetic order? or as per important column?

File: gcp/plugin.go (hunk 1 above)
Comment (ParthaI) [reply gcp/plugin.go]: Indentation fixed

File: gcp/table_gcp_bigquery_table.go [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+	}
+	return resp, err
+}
+
+//// TRANSFORM FUNCTIONS
+func gcpBigqueryTableAkas(ctx context.Context, h *transform.TransformData) (interface{}, error) {
+	data := TableID(h.HydrateItem)
+	projectID := strings.Split(data, ":")[0]
+	id := strings.Split(data, ":")[1]
+	akas := []string{"gcp://bigquery.googleapis.com/projects/" + projectID + "/tables/" + id}
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:205]: Can we include dataset details in `akas`?
e.g. `gcp://bigquery.googleapis.com/projects/<project_id>/datasets/<dataset_id>/tables/<table_id>`

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@ParthaI Added some suggestions. Please see comments, thanks!

File: gcp/table_gcp_bigquery_table.go (hunk 2 above)
Comment (Subhajit97) [reply gcp/table_gcp_bigquery_table.go:27]: @ParthaI We generally follow the below order
* For general columns
- First 4-5 will be important columns,
- All other columns will be added in alphabetical order,
- Add all the `JSON` type columns at the end (in alphabetical order)
* For standard columns
- Columns needs to be added in following order (i.e. `title`, `tags`(if applicable) and `akas`)

File: gcp/table_gcp_bigquery_table.go [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+				Name:        "expiration_time",
+				Description: "The time when this table expires, in milliseconds since the epoch. If not present, the table will persist indefinitely. Expired tables will be deleted and their storage reclaimed. The defaultTableExpirationMs property of the encapsulating dataset can be used to set a default expirationTime on newly created tables.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("ExpirationTime").Transform(transform.UnixMsToTimestamp),
+			},
+			{
+				Name:        "friendly_name",
+				Description: "A descriptive name for this table.",
+				Type:        proto.ColumnType_STRING,
+			},
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:60]: I think we should move this column at the top.

File: gcp/table_gcp_bigquery_table.go [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"dataset_id", "id"}),
+			Hydrate:    getBigqueryTable,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listBigQueryDatasets,
+			Hydrate:       listBigqueryTables,
+		},
+		Columns: []*plugin.Column{
+			// commonly used columns
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_table.go:26]:
```suggestion
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**